tests/test_butler.py

```diff
@@ -79,7 +79,7 @@ def testBasicPutGet(self):
         butler = Butler(self.configFile)
         # Create and register a DatasetType
         datasetTypeName = "test_metric"
-        dataUnits = ("camera", "visit")
+        dataUnits = ("Camera", "Visit")
```

Comment:
should they be case sensitive?

Reply:
I've been wondering about this, and I think the best approach is probably to be case (and underscore) insensitive, at least for data ID dictionary keys - the same words appear as both DataUnit type names (upper/camel case) and database field names (lower/snake case), and I think it'll be frustrating if we don't accept both in the dictionaries where the keys effectively refer to both.

That's probably out of scope for this ticket, though.